### PR TITLE
Add gutenboarding import module

### PR DESCRIPTION
**Pull request by bogiii**
#### Changes proposed in this Pull Request
Changes introduce a new Gutenboarding step named "import". The step lives on `/new/import` path. In this stage, idea is to provide UI without business logic. There is a `gutenboarding/import` feature flag which is only turn on for the `development` and `wpcalypso` environments.

For now, available import steps are:
- `List`, query param `?step=list`
- `Capture`, query param `?step=capture`
- `Scanning`, query param `?step=scanning`
- `Ready`, can be reached by adding query param `?step=ready`
- `Ready - No URL`, query param `?step=ready-no-url`
- `Not Ready`, query param `?step=ready-not`

**Desired Import Flow:**
Finish Signup -> Intent Screen -> Click "Import your site content" -> List / Capture / Scanning / Ready screens

#### Testing instructions
Manual testing:
- turn on `gutenboarding/import` feature from [config/development.json](https://github.com/Automattic/wp-calypso/blob/add/gutenboarding-import/config/development.json)
- go to the path `/new/import?step=`
- resize the screen to see how it looks on mobile

Related to #
- [Figma design](https://www.figma.com/file/TAK5FpwTvfbENJDfHc7Aq1/Site-Importing?node-id=1930%3A21706)

### CODE REVIEW

File: client/landing/gutenboarding/onboarding-block/import/ready/index.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+								),
+								website,
+								platform
+							),
+							{ strong: createElement( 'strong' ) }
+						) }
+					</SubTitle>
+
+					<div className="import__buttons-group">
+						<NextButton>{ __( 'Import your content' ) }</NextButton>
```
Comment (a8ci18n) [inline client/landing/gutenboarding/onboarding-block/import/ready/index.tsx:47]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **20** times:
`translate( 'Import Your Content' )` *ES Score: 11*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/gutenboarding-import)

File: client/landing/gutenboarding/onboarding-block/import/ready/platform-details.tsx [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+						{ createInterpolateElement( __( 'Things we <strong>can</strong> import:' ), {
+							strong: createElement( 'strong' ),
+						} ) }
+					</p>
+					<ul className={ 'import__details-list' }>
+						<li>
+							<Icon size={ 20 } icon={ check } /> { __( 'Blog posts' ) }
+						</li>
+						<li>
+							<Icon size={ 20 } icon={ check } /> { __( 'Static pages' ) }
```
Comment (a8ci18n) [inline client/landing/gutenboarding/onboarding-block/import/ready/platform-details.tsx:85]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **20** times:
`translate( 'All static pages' )` *ES Score: 7*

File: client/landing/gutenboarding/onboarding-block/import/index.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+		platform: 'Wix',
+	};
+
+	return (
+		<div className="gutenboarding-page import">
+			{ query.get( 'step' ) === 'ready' && (
+				<ReadyStep platform={ data.platform } website={ data.website } />
+			) }
+			{ query.get( 'step' ) === 'ready-no-url' && (
+				<ReadyNoUrlStep platform={ data.platform } website={ data.website } />
```
Comment (vishnugopal) [inline client/landing/gutenboarding/onboarding-block/import/index.tsx]: Will `ReadyNoUrl` have a website prop?

File: client/landing/gutenboarding/onboarding-block/import/index.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+			) }
+			{ query.get( 'step' ) === 'ready-not' && (
+				<ReadyNotStep platform={ data.platform } website={ data.website } />
```
Comment (vishnugopal) [inline client/landing/gutenboarding/onboarding-block/import/index.tsx]: Will `ReadyNotStep` require any of these props?

**Review by vishnugopal — CHANGES_REQUESTED** [review]
Hey great work! The design looks sweet, and also responsive, nice!

Please:
* Add a few screenshots and cc @ollierozdarz so that he can take a quick look at this before merge.
* Fix the preview being generated stage, since that'll be the most common case: 

CleanShot 2021-10-07 at 16 02 05@2x

* Teamcity [has a pretty strange error](https://teamcity.a8c.com/buildConfiguration/calypso_Calypso_E2E_Playwright_mobile/6800684?) that will possibly go away on a fresh build:

```
  ● [WPCOM] Gutenboarding: Language: (mobile) @parallel › Change Language › Acquire intent is displayed in Hebrew
    expect(received).toBe(expected) // Object.is equality
    Expected: "האתר שלי נקרא"
    Received: "私のサイトの名前は"
      46 |     it( `Acquire intent is displayed in ${ language }`, async function () {
      47 |       const text = await gutenboardingFlow.getSiteTitleLabel();
    > 48 |       expect( text ).toBe( labelText );
         |                      ^
      49 |     } );
      50 |   } );
      51 | } );
```

* All the suggestion for the localization bot look like no-brainers, so please execute them!

File: client/landing/gutenboarding/onboarding-block/import/index.tsx (hunk 3 above)
Comment (bogiii) [reply client/landing/gutenboarding/onboarding-block/import/index.tsx]: Good catch, the ReadyNoUrl component doesn't need that prop. It was a copy/paste oversight.

Thanks!

File: client/landing/gutenboarding/onboarding-block/import/index.tsx (hunk 4 above)
Comment (bogiii) [reply client/landing/gutenboarding/onboarding-block/import/index.tsx]: Good catch, the ReadyNot component doesn't need any props. It was a copy/paste oversight.

Thanks!

**Comment by bogiii** [comment]
Hey @ollierozdarz,

You can check the current state of screens on the following links:
- https://container-fervent-gauss.calypso.live/new/import?step=ready
- https://container-fervent-gauss.calypso.live/new/import?step=ready-no-url
- https://container-fervent-gauss.calypso.live/new/import?step=ready-not
- https://container-fervent-gauss.calypso.live/new/import?step=scanning
- https://container-fervent-gauss.calypso.live/new/import?step=list

cc @vishnugopal

File: client/landing/gutenboarding/onboarding-block/import/list/index.tsx [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+const ListStep: React.FunctionComponent = () => {
+	const { __ } = useI18n();
+
+	return (
+		<>
+			<div className={ 'import-layout list__wrapper' }>
+				<div className={ 'import-layout__column' }>
+					<div className="import__heading">
+						<Title>{ __( 'Import your content from another platform' ) }</Title>
```
Comment (a8ci18n) [inline client/landing/gutenboarding/onboarding-block/import/list/index.tsx:20]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **17** times:
`translate( 'Import content from another website or platform.' )` *ES Score: 7*

File: client/landing/gutenboarding/onboarding-block/import/capture/index.tsx [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+	const { __ } = useI18n();
+
+	return (
+		<div className="capture__content">
+			<input
+				className="capture__input"
+				autoComplete="off"
+				autoCorrect="off"
+				spellCheck="false"
+				placeholder={ __( 'Enter your site address' ) }
```
Comment (a8ci18n) [inline client/landing/gutenboarding/onboarding-block/import/capture/index.tsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **21** times:
`translate( 'Enter your new site address' )` *ES Score: 7*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/gutenboarding-import)

File: client/landing/gutenboarding/onboarding-block/import/capture/index.tsx [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+	return (
+		<div className="import-layout__center">
+			<div className="capture__content">
+				<input
+					className="capture__input"
+					autoComplete="off"
+					autoCorrect="off"
+					spellCheck="false"
+					placeholder={ __( 'Enter your site address' ) }
```
Comment (a8ci18n) [inline client/landing/gutenboarding/onboarding-block/import/capture/index.tsx:46]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **21** times:
`translate( 'Enter your new site address' )` *ES Score: 7*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/gutenboarding-import)

**Comment by vishnugopal** [comment]
Just a note that we are deprecating this PR in [favor of these ones](https://github.com/Automattic/wp-calypso/issues/57131).

**Pull request closed without merging**